src/main/java/com/grimacegram/grimacegram/controllers/UserController.java

```diff
@@ -0,0 +1,82 @@
+package com.grimacegram.grimacegram.controllers;
+
+import com.fasterxml.jackson.annotation.JsonView;
+import com.grimacegram.grimacegram.error.ApiError;
+import com.grimacegram.grimacegram.model.User;
+import com.grimacegram.grimacegram.repository.UserRepository;
+import com.grimacegram.grimacegram.services.UserService;
+import com.grimacegram.grimacegram.shared.GenericResponse;
+import com.grimacegram.grimacegram.shared.Views;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+import java.util.HashMap;
+import java.util.Map;
+
+@RestController
+@RequestMapping("/api/1.0")
+public class UserController {
+    @Autowired
+    UserService userService;
+
+
+    @PostMapping("/users")
+    GenericResponse createUser(@Valid @RequestBody User user){
```

Comment:
It would be better to use a custom UserDTO class with fields required for User creation here and accept it a parameter in `userService.save` method. You don't need to use the same class everywhere. Creating custom built dto's is better.

Reply:
ok, lets discuss it in our meeting?